src/main/java/com/epam/java/training/controller/ProductController.java

```diff
@@ -0,0 +1,286 @@
+package com.epam.java.training.controller;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.epam.java.training.entity.Product;
+import com.epam.java.training.service.IProductService;
+import com.epam.java.training.vo.Review;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+@Controller
+@Api(tags={"Product Service API"})
+public class ProductController {
+	protected Logger logger = LoggerFactory.getLogger(this.getClass());
+	@Autowired
+	private IProductService productService;
+
+	@GetMapping("products/{id}")
+	@ApiOperation(value="Product Information", notes="Provides the Product Information for a given productId", response=ResponseEntity.class)
+	@ApiImplicitParams({
+        @ApiImplicitParam(name = "id", value = "Product Id", required=true, dataType="Integer", paramType="path")
+	})
+	@ApiResponses(value={
+			@ApiResponse(code=200, message="Success"),
+			@ApiResponse(code=401, message="Bad Request"),
+			@ApiResponse(code=402, message="Unauthorised"),
+			@ApiResponse(code=403, message="Forbidden"),
+			@ApiResponse(code=404, message="No Product is found"),
+			@ApiResponse(code=500, message="Internal Server Error")
+	})
+	public ResponseEntity<Product> getProductById(@PathVariable("id") Integer id) {
+		logger.info("Before Review response received====....");
+		Product product = productService.getProductById(id);
+
+		HttpHeaders headers = new HttpHeaders();
+		headers.setContentType(MediaType.APPLICATION_JSON);
+		RestTemplate restTemplate = new RestTemplate();
```

Comment:
Define RestTemplate  bean in your configuration and inject it this class instead of creating it in every method

Reply:
Completed